omnigibson/scenes/scene_base.py

```diff
@@ -197,6 +197,11 @@ def _load_objects_from_scene_file(self):
         init_info = scene_info["objects_info"]["init_info"]
         init_state = scene_info["state"]["object_registry"]
         init_systems = scene_info["state"]["system_registry"].keys()
+        task_metadata = {}
+        try:
```

Comment:
is there a cleaner way to do this? I think the more principled way to do this is:

- Have a general API in the sim class that allows any external object to add a metadata callback function
- Have the Env class register the metadata callback function to cache the task metadata appropriately

vs. hardcoding these lines in the scene class.What do you think?

Reply:
Agreed that this does not look ideal upon first look, but moving this to the task would mean the scene would need a handle to it or that the metadata would need to be passed down here, which I think is overkill. Keeping as-is as agreed offline.